cirq/protocols/resolve_parameters.py

```diff
@@ -97,3 +97,44 @@ def resolve_parameters(
         return result
     else:
         return val
+
+
+def check_parameters(val: Any, param_resolver: 'cirq.study.Sweepable') -> Any:
+    """Checks for unresolved parameters.
+
+    This function will check the provided circuit against the
+    provided parameters to determined if there are any unresolved
+    parameters.
+
+    Args:
+        val: The object to check for missing parameters
+        param_resolver: the object to use for resolving all symbols
+
+    Returns:
+        Any unresolved parameters.
+    """
+    from cirq import protocols, study
+
+    try:
+        all_ops = list(val.all_operations())
+    except TypeError:
+        all_ops = []
+    c_params = []  # type: List[str]
+    s_params = []  # type: List[str]
+    param_list = study.to_resolvers(param_resolver)
+
+    for x in range(len(all_ops)):
+        if is_parameterized(all_ops[x]):
+            c_params.append(
+                str(protocols.circuit_diagram_info(all_ops[x]).exponent))
+
+    for param in param_list:
+        if hasattr(param, 'param_dict'):
```

Comment:
Similar to my question on ensuring that I am capturing all of the unparameterized objects, are there other examples for how the parameters might be represented? I made a number of tweaks to this section of code to ensure that the existing tests were passed, though if there are other examples that I should consider, it would be very much appreciated if you could please help me identify what those might be.